src/utils/graph_factory.hpp

```diff
@@ -124,6 +140,196 @@ template<typename G> void GraphFactory::copy_to(G& g) const
+{
+    
+  using v_new_desc_t = typename boost::graph_traits<G>::vertex_descriptor;
+  using e_new_desc_t = typename boost::graph_traits<G>::edge_descriptor;
+
+
+  if constexpr (has_reserve_for_vertex_list<G>::value) {
+    g.m_vertices.reserve(m_g.m_vertices.size());
+  }
+
+  using directed_category = typename boost::graph_traits<G>::directed_category;
+  constexpr bool is_bidirectional
+    = std::is_same<directed_category, boost::bidirectional_tag>::value;
+
+  
+  libsbml::ListOfReactions* reactionslist = model.getListOfReactions();
+
+  if  (reactionslist == nullptr) {
+    WCS_THROW("Invalid reactionslist pointer");
+    return; 
+  }
+  unsigned int reactionsSize = reactionslist->size();
+  libsbml::ListOfSpecies* specieslist = model.getListOfSpecies();
+
+  using species_added = std::unordered_map<std::string, v_new_desc_t>;
+  typename species_added::const_iterator it;
+  species_added smap;
+
+  using edges_added = std::unordered_map<std::string, e_new_desc_t>;
+  typename edges_added::const_iterator eit;
+  edges_added emap;
+
+  using undeclared_reactants = std::unordered_set<std::string>;
+  typename undeclared_reactants::const_iterator urit;
+  undeclared_reactants urset;
+
+  using  all_species = std::unordered_set<std::string>;
+  typename all_species::const_iterator aspit;
+  all_species aspset;
+  
+  // Create an unordered_set for all model species
+  unsigned int speciesSize = specieslist->size();
+  for (unsigned int si = 0u; si < speciesSize; si++) {
+    aspset.insert(specieslist->get(si)->getIdAttribute());
+  }
+
+  /// Add reactions
+  for (unsigned int ri=0u; ri < reactionsSize; ri++) {
+    if (reactionslist->get(ri) == nullptr) {
+      WCS_THROW("Invalid reaction pointer");
+      return;
+    } 
+    const auto &reaction = *(reactionslist->get(ri));
+        
+    wcs::Vertex v(model, reaction, g);
+    
+    v_new_desc_t vd = boost::add_vertex(v, g);
+    unsigned int reactantsSize = reaction.getNumReactants();
+    unsigned int productsSize = reaction.getNumProducts();
+    if constexpr (has_reserve_for_vertex_out_edges<G>::value) {
+        g.m_vertices[vd].m_out_edges.reserve(productsSize);
+    }
+    if constexpr (has_reserve_for_vertex_in_edges<G>::value && is_bidirectional) {
+      g.m_vertices[vd].m_in_edges.reserve(reactantsSize);
+    }
+
+    
+    /// Add reactants species
+    for (unsigned int si=0u; si < reactantsSize; si++) {
+      const auto &reactant = *(reaction.getReactant(si));
+      
+      std::string s_label = specieslist->get(reactant.getSpecies())->getIdAttribute();
+      it = smap.find(s_label) ;
+      v_new_desc_t vds;
+      if (it == smap.end()) {
+        wcs::Vertex vs(*specieslist->get(reactant.getSpecies()), g); 
+        vds = boost::add_vertex(vs, g); 
+        smap.insert(std::make_pair(s_label,vds));
+      } else {
+        vds = it->second;
+      }
+
+      std::string e_label = g[vds].get_label() + g[vd].get_label();
```

Comment:
g[vds].get_label() + '|' + g[vd].get_label()
Better to have a separator between two species names. We could have used '-'. However, in that case it can be confusing if it is a part of the species name itself. We just hope no sane person use '|' in a species name.